### PR TITLE
refactor event id

### LINKED ISSUE #965: Get rid of HostComponentLogger
The build function should take aggregator | operator. That defines which store/linker we use instead of depending on the type of log function. We don't even need separate log functions (we have separate log functions now, because we have separate bindings). Should come after #964

### LINKED ISSUE #1080: Follow correct EventId semantics
See https://github.com/Lay3rLabs/Layer-Wiki/wiki/Signed-Envelope#event-id

### PR DESCRIPTION
**Pull request by dakom**
* closes #1080 
* effectively closes #965
* depends on WIT in https://github.com/Lay3rLabs/wavs-wasi/pull/92

### CODE REVIEW

File: packages/engine/src/bindings/types/component_to_wavs.rs [hunk 1]
```diff
@@ -265,27 +265,13 @@ impl From<component_service::SignaturePrefix> for wavs_types::SignaturePrefix {
...
+impl From<component_output::WasmResponse> for wavs_types::WasmResponse {
+    fn from(src: component_output::WasmResponse) -> Self {
+        Self {
             payload: src.payload,
             ordering: src.ordering,
-            event_id,
-        })
+            event_id_salt: src.event_id_salt,
+        }
     }
```
Comment (ueco-jb) [inline packages/engine/src/bindings/types/component_to_wavs.rs]: Since the salt comes from the component should we include at least length checks on it?

File: packages/engine/src/bindings/operator/host.rs [hunk 2]
```diff
@@ -57,8 +57,16 @@ impl super::world::host::Host for OperatorHostComponent {
...
-    fn get_event_id(&mut self) -> Vec<u8> {
-        self.event_id.as_ref().to_vec()
+    fn get_event_id(&mut self, salt: Option<Vec<u8>>) -> Vec<u8> {
+        let salt = match salt.as_ref() {
+            Some(salt) => EventIdSalt::WasmResponse(salt),
+            None => EventIdSalt::Trigger(&self.trigger_data),
+        };
+
+        wavs_types::EventId::new(&self.service.id(), &self.workflow_id, salt)
+            .unwrap() // very unlikely to happen, would be a bincode error
```
[the remark this answers is not in the document]
Comment (dakom) [reply packages/engine/src/bindings/operator/host.rs]: the error that will be unwrapped is the bincode encode error itself

**Review by ueco-jb — COMMENTED** [review]
I would suggest adding some edge cases to EventId tests (an empty salt, a large one).

File: wit-definitions/operator/wit/operator.wit [hunk 3]
```diff
@@ -30,11 +30,10 @@ interface output {
         payload: list<u8>,
         // currently unused
         ordering: option<u64>,
+        // if not supplied, this will be `trigger-data`
```
[the remark this answers is not in the document]
Comment (dakom) [reply wit-definitions/operator/wit/operator.wit]: not quite, it actually _is_ the trigger data. The hash is `hash(ServiceId || WorkflowId || Salt)`, where salt is either what they provide or the trigger data itself
